crucible-llvm/src/Lang/Crucible/LLVM/Intrinsics/Libc.hs

```diff
@@ -495,6 +523,29 @@ callRealloc sym mvar alignment (regValue -> ptr) (regValue -> sz) =
        ]
 
 
+callPosixMemalign
+  :: (IsSymInterface sym, HasPtrWidth wptr, wptr ~ ArchWidth arch, ?lc :: TypeContext)
+  => sym
+  -> GlobalVar Mem
+  -> RegEntry sym (LLVMPointerType wptr)
+  -> RegEntry sym (BVType wptr)
+  -> RegEntry sym (BVType wptr)
+  -> OverrideSim p sym (LLVM arch) r args ret (RegValue sym (BVType 32))
+callPosixMemalign sym mvar (regValue -> outPtr) (regValue -> align) (regValue -> sz) =
+  case asUnsignedBV align >>= toAlignment . toBytes of
+    Nothing -> fail $ unwords ["posix_memalign: invalid alignment value:", show (printSymExpr align)]
```

Comment:
Does this indicate an invalid alignment value? Or a symbolic one? It seems like this is a case like `strlen` where we can't handle symbolic data for more fundamental reasons.

Reply:
Yeah, this requires a concrete alignment value.  We reject if the value is symbolic, or if it isn't a power of 2.  I suppose we could have more fine-grained error reporting for the two different situations.

Reply:
I think that might be nice, because if someone sees this due to a symbolic alignment, it implies that they are doing something wrong as a user of Crucible (and perhaps that they could fix it), whereas the situation is less tractable than that.